content/docs/reference-react.md

```diff
@@ -212,136 +211,136 @@ You will not typically invoke `React.createFactory()` directly if you are using
+* [Przekazywanie referencji komponentom wyższego rzędu](/docs/forwarding-refs.html#forwarding-refs-in-higher-order-components)
 
-`React.forwardRef` accepts a rendering function as an argument. React will call this function with `props` and `ref` as two arguments. This function should return a React node.
+`React.forwardRef` przyjmuje funkcję renderującą jako argument. React wywoła tę funkcję biorąc `props` oraz `ref` jako dwa argumenty. Funkcja ta powinna zwrócić węzeł reaktowy.
 
 `embed:reference-react-forward-ref.js`
 
-In the above example, React passes a `ref` given to `<FancyButton ref={ref}>` element as a second argument to the rendering function inside the `React.forwardRef` call. This rendering function passes the `ref` to the `<button ref={ref}>` element.
+W powyższym przykładzie, React przekazuje `ref` będąc dla `<FancyButton ref={ref}>` drugim argumentem do funkcji renderującej wewnątrz `React.forwardRef`. Funkcja renderująca przekaże `ref` do elementu `<button ref={ref}>`.
 
-As a result, after React attaches the ref, `ref.current` will point directly to the `<button>` DOM element instance.
+W rezultacie po tym, jak referencja zostanie przypięta przez Reacta, `ref.current` będzie wskazywał bezpośrednio na element `<button>`.
 
-For more information, see [forwarding refs](/docs/forwarding-refs.html).
+Po więcej informacji zobacz [przekazywanie referencji](/docs/forwarding-refs.html).
 
 ### `React.lazy` {#reactlazy}
 
-`React.lazy()` lets you define a component that is loaded dynamically. This helps reduce the bundle size to delay loading components that aren't used during the initial render.
+`React.lazy()` pozwala Ci zdefiniować komponenty, które są ładowane dynamicznie. Pozwala to zredukować rozmiar paczki, poprzez wyodrębnienie komponentów, które nie są używane podczas inicjalizacji renderowania.
 
-You can learn how to use it from our [code splitting documentation](/docs/code-splitting.html#reactlazy). You might also want to check out [this article](https://medium.com/@pomber/lazy-loading-and-preloading-components-in-react-16-6-804de091c82d) explaining how to use it in more detail.
+W naszej [dokumentacji poświęconej rozdzielaniu kodu](/docs/code-splitting.html#reactlazy) zamieściliśmy więcej informacji na ten temat. [Ten artykuł](https://medium.com/@pomber/lazy-loading-and-preloading-components-in-react-16-6-804de091c82d) również może okazać się pomocny w wyjaśnieniu szczegółów działania tej metody.
 
 ```js
-// This component is loaded dynamically
+// Ten komponent jest ładowany dynamicznie
 const SomeComponent = React.lazy(() => import('./SomeComponent'));
 ```
 
-Note that rendering `lazy` components requires that there's a `<React.Suspense>` component higher in the rendering tree. This is how you specify a loading indicator.
+Pamiętaj, że renderowanie leniwych komponentów (ang. *lazy*) wymaga użycia komponentu `<React.Suspense>` wyżej w drzewie renderowania. W ten sposób definiujesz wskaźnik ładowania.
```

Comment:
```suggestion
Pamiętaj, że renderowanie "leniwych komponentów" (ang. *lazy components*) wymaga użycia komponentu `<React.Suspense>` na wyższym poziomie drzewa. W ten sposób definiuje się wskaźnik ładowania.
```
Nie do końca pasuje mi tu ostatnie zdanie, ale w angielskiej wersji też jest to dość niejasno napisane...